### ISSUE TITLE
Periodic crashes

### ISSUE THREAD

**Issue by andrew-boyarshin** [dremin/RetroBar#24]
Runtime:
```
.NET SDK (reflecting any global.json):
 Version:   6.0.100-preview.7.21318.10
 Commit:    4e374bb055

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.21390
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100-preview.7.21318.10\

Host (useful for support):
  Version: 6.0.0-preview.7.21318.2
  Commit:  140120290a

.NET SDKs installed:
  6.0.100-preview.5.21302.13 [C:\Program Files\dotnet\sdk]
  6.0.100-preview.7.21318.10 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.0-preview.5.21301.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0-preview.7.21318.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0-preview.5.21301.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0-preview.7.21318.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.0-preview.5.21301.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.0-preview.7.21316.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

Custom built RetroBar at c3708fe564771c62bb48fe2fa7eaf39c93c7b993 in Release mode after retargetting to `net6.0-windows10.0.19041.0`.

WinDbg:
```
0:000> !CLRStack -a
OS Thread Id: 0x2384 (0)
        Child SP               IP Call Site
000000A96F57E140 00007ff8bb9098bb [InlinedCallFrame: 000000a96f57e140] ManagedShell.Interop.NativeMethods.SHUnlockShared()
000000A96F57E140 00007ff85bc91cf0 [InlinedCallFrame: 000000a96f57e140] ManagedShell.Interop.NativeMethods.SHUnlockShared()
000000A96F57E110 00007ff85bc91cf0 ILStubClass.IL_STUB_PInvoke(IntPtr)
    PARAMETERS:

000000A96F57E1D0 000001f436353c2e ManagedShell.WindowsTasks.ApplicationWindow.SetOverlayIconDescription()

000000A96F57E230 00007ff85c97694c ManagedShell.WindowsTasks.TasksService.ShellWinProc()

000000A96F57E350 00007ff85c92b474 ManagedShell.Common.SupportingClasses.NativeWindowEx.WndProc()

000000A96F57E3B0 00007ff85c92b263 System.Windows.Forms.NativeWindow.Callback(IntPtr, WM, IntPtr, IntPtr) [/_/src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs @ 400]
    PARAMETERS:
        this = <no data>

000000A96F57E420 00007ff85c2307f0 ILStubClass.IL_STUB_ReversePInvoke(Int64, UInt32, Int64, Int64)
    PARAMETERS:

000000A96F57E7E8 00007ff91f2cdd5c [InlinedCallFrame: 000000a96f57e7e8] MS.Win32.UnsafeNativeMethods.IntGetMessageW(System.Windows.Interop.MSG ByRef, System.Runtime.InteropServices.HandleRef, Int32, Int32)
000000A96F57E7E8 00007ff85c4330a1 [InlinedCallFrame: 000000a96f57e7e8] MS.Win32.UnsafeNativeMethods.IntGetMessageW(System.Windows.Interop.MSG ByRef, System.Runtime.InteropServices.HandleRef, Int32, Int32)
000000A96F57E7C0 00007ff85c4330a1 ILStubClass.IL_STUB_PInvoke(System.Windows.Interop.MSG ByRef, System.Runtime.InteropServices.HandleRef, Int32, Int32)
    PARAMETERS:

000000A96F57E880 00007ff85c962921 MS.Win32.UnsafeNativeMethods.GetMessageW(System.Windows.Interop.MSG ByRef, System.Runtime.InteropServices.HandleRef, Int32, Int32)
    PARAMETERS:

000000A96F57E8C0 00007ff85c92ad2d System.Windows.Threading.Dispatcher.GetMessage(System.Windows.Interop.MSG ByRef, IntPtr, Int32, Int32)
    PARAMETERS:
        this = <no data>

000000A96F57E950 00007ff877da9548 System.Windows.Threading.Dispatcher.PushFrameImpl(System.Windows.Threading.DispatcherFrame)
    PARAMETERS:
        this (0x000000A96F57E9E0) = 0x000001f4141112d0

000000A96F57E9E0 00007ff877da6f31 System.Windows.Threading.Dispatcher.PushFrame(System.Windows.Threading.DispatcherFrame)
    PARAMETERS:

000000A96F57EA10 00007ff877da6ede System.Windows.Threading.Dispatcher.Run()

000000A96F57EA40 00007ff875fb048c System.Windows.Application.RunDispatcher(System.Object)
    PARAMETERS:
        this = <no data>

000000A96F57EA70 00007ff875faeb3f System.Windows.Application.RunInternal(System.Windows.Window)
    PARAMETERS:
        this (<CLR reg>) = 0x000001f4140eefa8

000000A96F57EAC0 00007ff875fabc3c System.Windows.Application.Run()
    PARAMETERS:
        this = <no data>

000000A96F57EAF0 000000014000d590 RetroBar.App.Main()

0:000> !EHInfo
0:000> .exptr 0x000000a96f57c9f0

----- Exception record at 000000a9`6f57ca00:
ExceptionAddress: 00007ff8bb879c6f (coreclr!ProcessCLRException+0x00000000000e165f)
   ExceptionCode: c0000005 (Access violation)
  ExceptionFlags: 00000001
NumberParameters: 0

----- Context record at 000000a9`6f57caa0:
rax=000000a96f57caa0 rbx=00000000c0000005 rcx=000000a96f57caa0
rdx=0000000000000000 rsi=00007ff8bb879c6f rdi=0000000000000000
rip=00007ff8bb9098bb rsp=000000a96f57c9c0 rbp=000000a96f57cac0
 r8=00000000000004d0  r9=0000000000000000 r10=0000000000000000
r11=000000a96f57caa0 r12=0000000000000000 r13=000001f413eb45a0
r14=000001f43fdb83d0 r15=0000000000000000
iopl=0         nv up ei pl nz na pe nc
cs=0033  ss=002b  ds=002b  es=002b  fs=0053  gs=002b             efl=00000202
coreclr!EEPolicy::HandleFatalError+0x7f:
00007ff8`bb9098bb 488d442440      lea     rax,[rsp+40h]
```

**Comment by dremin** [comment]
Thanks for the report! I am pretty sure I have a fix for this, I'll update here once it is pushed.

**Comment by dremin** [comment]
> 
> 
> Thanks for the report! I am pretty sure I have a fix for this, I'll update here once it is pushed.

@andrew-boyarshin Could you please try with the latest commit?

**Comment by andrew-boyarshin** [comment]
I've been rocking the latest commit for 4 hours now, no crashes. Thanks! Closing as fixed by 4bfc13ad5605b0f564a40c645dbbe35e0e4d5402 and cairoshell/ManagedShell#27.
